### PR TITLE
Feature/add support to new meta in detectiocrop

### PR DESCRIPTION
**Pull request by tvlenin**

### CODE REVIEW

File: gst/inferencecrop/gstdetectioncrop.cc [hunk 1]
```diff
@@ -348,127 +291,163 @@ gst_detection_crop_set_caps (GstPad *pad, GParamSpec *unused,
...
     GST_LOG_OBJECT (self, "No meta found, dropping buffer");
     goto out;
   }
 
-  if (meta->num_boxes <= 0) {
-    GST_LOG_OBJECT (self, "Meta has no valid objects");
-    goto out;
-  }
+  num_detections = 0;
+  enabled_detections = gst_detection_crop_find_predictions (self, &num_detections,
```
[the remark this answers is not in the document]
Comment (rrcarlosrodriguez) [reply gst/inferencecrop/gstdetectioncrop.cc]: It looks like what you need here is to know if there is at least 1 prediction enabled right? Why don't you use something like this: https://github.com/RidgeRun/gst-inference/blob/feature/change-collectpads-approach/gst-libs/gst/r2inference/gstvideoinference.c#L1052

[the remark this answers is not in the document]
Comment (tvlenin) [reply gst/inferencecrop/gstdetectioncrop.cc]: Actually the return value is not necessary, I converted it to void.

File: gst/inferencecrop/gstdetectioncrop.cc [hunk 2]
```diff
@@ -348,127 +293,156 @@ gst_detection_crop_set_caps (GstPad *pad, GParamSpec *unused,
...
-                                   meta);
-  if (-1 == requested_index) {
-    goto out;
+  num_detections = 0;
+  gst_detection_crop_find_predictions (self, &num_detections,
+                       inference_meta, &list, inference_meta->prediction);
+
+  for (iter = list; iter != NULL; iter = g_list_next(iter)) {
+    GstInferencePrediction *pred = (GstInferencePrediction *)iter->data;
+    if ( 0 != pred->prediction_id) {
```
Comment (michaelgruner) [inline gst/inferencecrop/gstdetectioncrop.cc]: Why is this check required? I believe you already discarded the root.

File: gst/inferencecrop/gstdetectioncrop.cc [hunk 3]
```diff
@@ -348,127 +293,156 @@ gst_detection_crop_set_caps (GstPad *pad, GParamSpec *unused,
...
+      GST_LOG_OBJECT (self, "BBox: %dx%dx%dx%d", box.x, box.y, box.width,
+                      box.height);
+
+      gst_detection_crop_new_buffer_size (self, box.x, box.y, box.width, box.height,
+                                          crop_width_ratio, crop_height_ratio,
+                                          &top, &bottom, &right, &left);
+      self->element->SetCroppingSize ((gint) top, (gint) bottom, (gint) right,
+                                      (gint) left);
+
+      croped_buffer = gst_buffer_copy_deep (buffer);
```
Comment (michaelgruner) [inline gst/inferencecrop/gstdetectioncrop.cc]: It is not necessary to do a deep copy, we can reuse the data and only copy the GstBuffer.

File: gst/inferencecrop/gstdetectioncrop.cc [hunk 4]
```diff
@@ -348,127 +293,156 @@ gst_detection_crop_set_caps (GstPad *pad, GParamSpec *unused,
...
+      dmeta = (GstInferenceMeta *) gst_buffer_add_meta (croped_buffer,
+              GST_INFERENCE_META_INFO,
+              NULL);
+
+      pred->bbox.x = 0;
```
Comment (michaelgruner) [inline gst/inferencecrop/gstdetectioncrop.cc]: You are modifying the original prediction, hence, corrupting it. Modify the copy.

**Comment by rrcarlosrodriguez** [comment]
Use the following license header for new files only:

```
/*
 * GStreamer
 * Copyright (C) 2019 RidgeRun <support@ridgerun.com>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Library General Public
 * License as published by the Free Software Foundation; either
 * version 2 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Library General Public License for more details.
 *
 * You should have received a copy of the GNU Library General Public
 * License along with this library; if not, write to the
 * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 *
 */
```

**Pull request merged**